Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -280,5 +293,171 @@ TimeROI TimeSplitter::getTimeROI(const int workspaceIndex) {
+ * - Pulse + TOF time for each event, when pulseTof==true && tofCorrect==false
+ * - Pulse + (factor * TOF + shift) for each event, when pulseTof==true && tofCorrect==true
+ *
+ * @param events : list of input events
+ * @param pulseTof : if True, split according to Pulse + TOF time, otherwise split by Pulse time
+ * @param tofCorrect : rescale and shift the TOF values (factor*TOF + shift)
+ * @param factor : rescale the TOF values by a dimensionless factor.
+ * @param shift : shift the TOF values after rescaling, in units of microseconds.
+ * @return : times that will be compared again the splitter boundaries, in units of nanoseconds.
+ */
+std::vector<int64_t> TimeSplitter::sortEventList(const EventList &events, bool pulseTof, bool tofCorrect, double factor,
+                                                 double shift) const {
+  // sort the list in-place
+  if (pulseTof)
+    // this sorting is preserved under linear transformation tof --> factor*tof+shift with factor>0
+    events.sortPulseTimeTOF();
+  else
+    events.sortPulseTime();
+
+  // extract the times we'll be comparing against the splitter times
+  std::vector<int64_t> sortedTimes;
+  auto pulseTimes = events.getPulseTimes();
+  for (auto pulseTime : pulseTimes)
+    sortedTimes.emplace_back(pulseTime.totalNanoseconds());
+  if (pulseTof) {
+    auto tofs = events.getTofs(); // units of microseconds
+    if (tofCorrect)               // modify tofs in-place
+      std::for_each(tofs.begin(), tofs.end(), [&](double &tof) { tof = factor * tof + shift; });
+    constexpr double micro2nanosec{1000}; // from microsecond to nanosecond
+    for (size_t i = 0; i < sortedTimes.size(); i++)
+      sortedTimes[i] += static_cast<int64_t>(tofs[i] * micro2nanosec); // add TOF to pulse time
+  }
+
+  return sortedTimes;
+}
+
+/**
+ * Distribute a list of events by comparing a vector of times against the splitter boundaries.
+ *
+ * Each event in `events` has a corresponding time in `times`, which we use to find a destination index
+ * in the TimeSplitter object. The destination index is the key to find the target event list
+ * in the partials map.
+ *
+ * @tparam EVENTTYPE : one of EventType::TOF or EventType::WEIGHTED
+ * @param times : times associated to the events, used to find the destination index
+ * @param events : list of input times
+ * @param partials : target list of partial event lists, associated to the different destination indexes
+ * @throws : if the size of times and events are different
+ */
+template <typename EVENTTYPE>
+void TimeSplitter::splitEventVec(const std::vector<int64_t> &times, const std::vector<EVENTTYPE> &events,
+                                 std::map<int, EventList *> partials) const {
+  assert(times.size() == events.size());
+  // initialize the iterator over the splitter
+  // it assumes the splitter keys (DateAndTime objects) are sorted by increasing time.
+  auto itSpl = m_roi_map.cbegin();                // iterator over the splitter
```

Comment:
The iterator variable names are a little confusing here, especially `itVec` which is an iterator for the events themselves. How about
* `itSpl` becomes `itSplitter` 
* `itTim` becomes `itTime` (one extra character helps a ton)
* `itVec` becomes `itEvent`